elixir/apps/web/lib/web/live/policies/new.ex

```diff
@@ -83,20 +86,37 @@ defmodule Web.Policies.New do
 
   def handle_event("validate", %{"policy" => policy_params}, socket) do
     form =
-      Policies.new_policy(policy_params, socket.assigns.subject)
+      policy_params
+      |> put_default_policy_params(socket)
+      |> Policies.new_policy(socket.assigns.subject)
       |> Map.put(:action, :validate)
       |> to_form()
 
     {:noreply, assign(socket, form: form)}
   end
 
   def handle_event("submit", %{"policy" => policy_params}, socket) do
+    policy_params = put_default_policy_params(policy_params, socket)
+
     with {:ok, policy} <- Policies.create_policy(policy_params, socket.assigns.subject) do
-      {:noreply, redirect(socket, to: ~p"/#{socket.assigns.account}/policies/#{policy}")}
+      if site_id = socket.assigns.params["site_id"] do
+        {:noreply,
+         push_navigate(socket, to: ~p"/#{socket.assigns.account}/sites/#{site_id}?#resources")}
+      else
+        {:noreply, redirect(socket, to: ~p"/#{socket.assigns.account}/policies/#{policy}")}
```

Comment:
Just curious for my own edification, was `redirect` chosen for a reason here?  Won't this do a whole page reload?

Reply:
Good catch. We have it in a ton of places, let's get this PR merged and I'll fix all the place in upcoming PR.

Reply:
Sounds good 👍